src/storage/cast.hpp

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Steinwurf ApS 2016.
+// All Rights Reserved
+//
+// Distributed under the "BSD License". See the accompanying LICENSE.rst file.
+
+/// @file storage.hpp This implementation was inspired by the
```

Comment:
This comment is in all the files, I think we can just move the "inspiration" part to the README.rst